Tools/TestWebKitAPI/Tests/WebKitCocoa/Download.mm

```diff
@@ -853,6 +853,36 @@ - (void)_download:(_WKDownload *)download didReceiveData:(uint64_t)length
 
 @end
 
+#if HAVE(MODERN_DOWNLOADPROGRESS)
+@interface ProgressObserver : NSObject
+- (instancetype)init;
+@end
+
+@implementation ProgressObserver {
+    bool receivedProgress;
+}
+
+- (instancetype)init
+{
+    if (!(self = [super init]))
+        return nil;
+    receivedProgress = false;
+    return self;
+}
+
+- (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NSDictionary *)change context:(void *)context
+{
+    receivedProgress = true;
```

Comment:
Could we check that the key path is as expected and the progress is more than 0% done?